### PR TITLE
fix(gatsby): regression ~ show non structured grapqhl errors

### LINKED ISSUE #15261: gatsby build ELIFECYCLE error code 1 with no error message
## Description

I'm having trouble building my gatsby app and I'm not able to find any error message to help debug the issue.

### `gatsby develop` works fine with no errors reported

`npx gatsby develop` -> success
`yarn develop` -> `gatsby develop` -> success
`npm run develop` -> `gatsby develop` -> success

### `gatsby build` works via `npx` but not `yarn` or `npm` even though it appears that all 3 are using the same version of gatsby 🤔 

`npx gatsby --version` and `yarn run gatsby --version` and the equivalent npm command all report:
```
Gatsby CLI version: 2.7.6
Gatsby version: 2.11.4
```

`npx gatsby build` -> success

`yarn build` -> `gatsby build` ->
```
$ gatsby build
success open and validate gatsby-configs - 0.017 s
success load plugins - 7.228 s
success onPreInit - 0.007 s
success delete html and css files from previous builds - 0.026 s
success initialize cache - 0.008 s
success copy gatsby files - 0.319 s
warn gatsby-plugin-feed was initialized in gatsby-config.js without a feeds option.
This means that the plugin will use the internal RSS feed creation, which may not
match your use case.
This behavior will be removed in the next major release of gatsby-plugin-feed.
success onPreBootstrap - 0.039 s
success source and transform nodes - 0.306 s
success building schema - 0.830 s
success createPages - 0.120 s
success createPagesStatefully - 0.038 s
success onPreExtractQueries - 0.007 s
success update schema - 0.048 s
success extract queries from components - 0.200 s
success write out requires - 0.006 s
success write out redirect data - 0.007 s
success Build manifest and related icons - 0.198 s
success onPostBootstrap - 0.213 s
⠀
info bootstrap finished - 24.217 s
⠀
success run static queries - 0.037 s — 2/2 61.52 queries/second
success Building production JavaScript and CSS bundles - 29.701 s
success Rewriting compilation hashes - 0.004 s
success run page queries - 0.352 s — 7/7 20.60 queries/second
success Building static HTML for pages - 3.958 s — 7/7 5.86 pages/second

error Command failed with exit code 1.
```

`npm run build` -> `gatsby develop` ->
```
> gatsby build

success open and validate gatsby-configs - 0.015 s
success load plugins - 3.653 s
success onPreInit - 0.010 s
success delete html and css files from previous builds - 0.031 s
success initialize cache - 0.010 s
success copy gatsby files - 0.258 s
warn gatsby-plugin-feed was initialized in gatsby-config.js without a feeds option.
This means that the plugin will use the internal RSS feed creation, which may not
match your use case.
This behavior will be removed in the next major release of gatsby-plugin-feed.
success onPreBootstrap - 0.040 s
success source and transform nodes - 0.143 s
success building schema - 0.783 s
success createPages - 0.117 s
success createPagesStatefully - 0.037 s
success onPreExtractQueries - 0.007 s
success update schema - 0.051 s
success extract queries from components - 0.147 s
success write out requires - 0.005 s
success write out redirect data - 0.002 s
success Build manifest and related icons - 0.150 s
success onPostBootstrap - 0.163 s
⠀
info bootstrap finished - 13.950 s
⠀
success run static queries - 0.042 s — 2/2 69.32 queries/second
success Building production JavaScript and CSS bundles - 12.646 s
success run page queries - 0.164 s — 7/7 43.97 queries/second
success Building static HTML for pages - 2.100 s — 7/7 9.31 pages/second

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! blog@0.1.0 build: `gatsby build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the blog@0.1.0 build script.
```

### Steps to reproduce

This is the repo: https://github.com/tripphamm/blog

You could try to pull it down and install/build.

### Environment

  System:
    OS: macOS 10.14.5
    CPU: (4) x64 Intel(R) Core(TM) i5-5250U CPU @ 1.60GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 8.10.0 - /usr/local/bin/node
    Yarn: 1.16.0 - ~/.yarn/bin/yarn
    npm: 6.4.1 - /usr/local/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 75.0.3770.100
    Safari: 12.1.1
  npmPackages:
    gatsby: ^2.3.25 => 2.11.4 
    gatsby-image: ^2.0.39 => 2.2.3 
    gatsby-mdx: ^0.6.3 => 0.6.3 
    gatsby-plugin-feed: ^2.1.1 => 2.3.1 
    gatsby-plugin-google-analytics: ^2.0.18 => 2.1.1 
    gatsby-plugin-manifest: ^2.0.29 => 2.2.0 
    gatsby-plugin-offline: ^2.0.25 => 2.2.0 
    gatsby-plugin-react-helmet: ^3.0.12 => 3.1.0 
    gatsby-plugin-sharp: ^2.0.35 => 2.2.1 
    gatsby-plugin-typography: ^2.2.13 => 2.3.0 
    gatsby-remark-copy-linked-files: ^2.0.11 => 2.1.0 
    gatsby-remark-images: ^2.0.6 => 2.0.6 
    gatsby-remark-prismjs: ^3.2.8 => 3.3.0 
    gatsby-remark-responsive-iframe: ^2.1.1 => 2.2.1 
    gatsby-remark-smartypants: ^2.0.9 => 2.1.0 
    gatsby-source-filesystem: ^2.0.29 => 2.1.1 
    gatsby-transformer-remark: ^2.3.12 => 2.5.0 
    gatsby-transformer-sharp: ^2.1.18 => 2.2.0 
  npmGlobalPackages:
    gatsby-cli: 2.4.8

### PR DESCRIPTION
**Pull request by wardpeet**
## Description
It looks like we're eating graphql errors not coming from createPages functions. I refactored graphql-runner a little bit so I could write a unit test for it.

## Related Issues
Fixes #15261

### CODE REVIEW

File: packages/gatsby/src/bootstrap/graphql-runner.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+const { graphql } = require(`graphql`)
+const stackTrace = require(`stack-trace`)
+const withResolverContext = require(`../schema/context`)
+const errorParser = require(`../query/error-parser`).default
+
+const createGraphqlRunner = (schema, reporter) => (query, context = {}) =>
+  graphql(
```
Comment (wardpeet) [inline packages/gatsby/src/bootstrap/graphql-runner.js]: we had `const schema = store.getState().schema` in graphqlRunner before. Unsure if we need to get the latest schema each time we run a graphql query. If not mistaken we don't modify the schema anymore.

Comment (sidharthachatterjee) [reply packages/gatsby/src/bootstrap/graphql-runner.js]: CC @freiksenet

Comment (pieh) [reply packages/gatsby/src/bootstrap/graphql-runner.js]: We do update schema once after createPages finished (to infer `SitePage.context`), I would leave this part as it was (getting schema from redux every time)

Comment (freiksenet) [reply packages/gatsby/src/bootstrap/graphql-runner.js]: What @pieh said.

**Pull request merged**